airflow/providers/asana/operators/asana_tasks.py

```diff
@@ -0,0 +1,170 @@
+    :type name: str
+    :param task_parameters: Any of the optional task creation parameters.
+        See https://developers.asana.com/docs/create-a-task for a complete list.
+        You must specify at least one of 'workspace', 'parent', or 'projects'
+        either here or in the connection.
+    :type task_parameters: dict
+    """
+
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        name: str,
+        task_parameters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.conn_id = conn_id
+        self.name = name
+        self.task_parameters = task_parameters
+
+    def execute(self, context: Dict) -> str:
+        hook = AsanaHook(conn_id=self.conn_id)
+        response = hook.create_task(self.name, self.task_parameters)
+        self.log.info(response)
+        return response["gid"]
+
+
+class AsanaUpdateTaskOperator(BaseOperator):
+    """
+    This operator can be used to update Asana tasks.
+    For more information on Asana optional task parameters, see
+    https://developers.asana.com/docs/update-a-task
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AsanaUpdateTaskOperator`
+
+    :param conn_id: The Asana connection to use.
+    :type conn_id: str
+    :param asana_task_gid: Asana task ID to update
+    :type asana_task_gid: str
+    :param task_parameters: Any task parameters that should be updated.
+        See https://developers.asana.com/docs/update-a-task for a complete list.
+    :type task_update_parameters: dict
+    """
+
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        asana_task_gid: str,
+        task_parameters: dict,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.conn_id = conn_id
+        self.asana_task_gid = asana_task_gid
+        self.task_parameters = task_parameters
+
+    def execute(self, context: Dict) -> None:
+        hook = AsanaHook(conn_id=self.conn_id)
+        response = hook.update_task(self.asana_task_gid, self.task_parameters)
+        self.log.info(response)
+
+
+class AsanaDeleteTaskOperator(BaseOperator):
+    """
+    This operator can be used to delete Asana tasks.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AsanaDeleteTaskOperator`
+
+    :param conn_id: The Asana connection to use.
+    :type conn_id: str
+    :param asana_task_gid: Asana Task ID to delete.
+    :type asana_task_gid: str
+    """
+
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        asana_task_gid: str,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.conn_id = conn_id
+        self.asana_task_gid = asana_task_gid
+
+    def execute(self, context: Dict) -> None:
+        hook = AsanaHook(conn_id=self.conn_id)
```

Comment:
In the connection you are able to set `workspace` and `project` but the operators doesn't seem to allow to overwrite the default. Does this means that we need to define connection per workspace,project? Should we allow overwrite in the operator level?

Reply:
Thanks for the comment, I can see how this could be confusing... you actually can override these (where applicable) in the dict used to pass optional Asana configuration parameters to the operator. For example, if you wanted to override the project defined in the connection when using the `AsanaFindTaskOperator`, you could pass `search_parameters={'project': 'some other project'}`. This is then passed to the `find_task` method of the hook, where `_merge_find_task_parameters` gets called and the project passed in through the operator's `search_parameters` will override the default project from the connection if present. I should probably add more documentation to make this clearer, though.

Reply:
How about Addiding that comment (or even example) in "example_asana.py" @jmelot . I promise that PR will be much faster :rocket: